packages/cli/src/commands/doctor/healthchecks/reactNativeGlobal.ts

```diff
@@ -78,82 +88,17 @@ const removeNodePackage = async (packageName, packageManager, loader) => {
 export default {
   label: label,
   getDiagnostics: async () => ({
-    needsToBeFixed: checkGlobalInstall(),
+    needsToBeFixed: !!checkGlobalInstallalations(),
   }),
   runAutomaticFix: async ({loader}) => {
     loader.stop();
 
-    let reactNativeYarnPath = '';
-    let reactNativeCLIYarnPath = '';
-    let reactNativeNpmPath = '';
-    let reactNativeCLINpmPath = '';
-    const reactNative = 'react-native';
-    const reactNativeCLI = 'react-native-cli';
-
-    //-----Get global paths for both yarn and npm-------
-    try {
-      reactNativeYarnPath = findGlobalPackage(
-        packageManagers.yarn,
-        reactNative,
-      );
-    } catch (error) {
-      return null;
-    }
-
-    try {
-      reactNativeCLIYarnPath = findGlobalPackage(
-        packageManagers.yarn,
-        reactNativeCLI,
-      );
-    } catch (error) {
-      return null;
-    }
-
-    try {
-      reactNativeNpmPath = findGlobalPackage(packageManagers.npm, reactNative);
-    } catch (error) {
-      return null;
-    }
-
-    try {
-      reactNativeCLINpmPath = findGlobalPackage(
-        packageManagers.npm,
-        reactNativeCLI,
-      );
-    } catch (error) {
-      return null;
-    }
-
-    //----Check if paths are in global directories; if so, remove---
-
-    // RN yarn and npm
-
-    if (isPathInside(reactNativeYarnPath, globalDirectories.yarn.packages)) {
-      await removeNodePackage(reactNative, 'yarn', loader);
-    }
-
-    if (
-      isPathInside(
-        reactNativeNpmPath,
-        fs.realpathSync(globalDirectories.npm.packages),
-      )
-    ) {
-      await removeNodePackage(reactNative, 'npm', loader);
-    }
-
-    // RNcli yarn and npm
-
-    if (isPathInside(reactNativeCLIYarnPath, globalDirectories.yarn.packages)) {
-      await removeNodePackage(reactNativeCLI, 'yarn', loader);
-    }
-
-    if (
-      isPathInside(
-        reactNativeCLINpmPath,
-        fs.realpathSync(globalDirectories.npm.packages),
-      )
-    ) {
-      await removeNodePackage(reactNativeCLI, 'npm', loader);
-    }
+    const globalPaths = checkGlobalInstallalations();
+
+    console.log('Global paths: ' + globalPaths);
+
+    await globalPaths.map(
+      async packagePath => await removeNodePackage(packagePath, loader),
```

Comment:
I believe you can omit `async/await` here.

```suggestion
      packagePath => removeNodePackage(packagePath, loader),
```